bson-record-codec/src/main/org/bson/codecs/record/RecordCodec.java

```diff
@@ -109,6 +115,36 @@ private static Codec<?> computeCodec(final RecordComponent component, final Code
             return codec;
         }
 
+        private static Class<?> resolveComponentType(final List<Type> typeParameters, final RecordComponent component) {
+            Type resolvedType = resolveType(component.getGenericType(), typeParameters, component.getDeclaringRecord());
+            return resolvedType instanceof Class<?> clazz ? clazz : component.getType();
+        }
+
+        private static List<Type> resolveActualTypeArguments(final List<Type> typeParameters, final Class<?> recordClass,
+                final ParameterizedType parameterizedType) {
+            return Arrays.stream(parameterizedType.getActualTypeArguments())
+                    .map(type -> resolveType(type, typeParameters, recordClass))
+                    .toList();
+        }
+
+        private static Type resolveType(final Type type, final List<Type> typeParameters, final Class<?> recordClass) {
+            return type instanceof TypeVariable<?> typeVariable
+                    ? typeParameters.get(getIndexOfTypeParameter(typeVariable.getName(), recordClass))
+                    : type;
+        }
+
+        // Get
+        private static int getIndexOfTypeParameter(final String typeParameterName, final Class<?> recordClass) {
```

Comment:
This looks a bit strange, but the reflection API requires that you match up type parameters by their index in the respective arrays.

Reply:
Thanks, I like this test.  

Done